dbms/src/Storages/DeltaMerge/StoragePool.cpp

```diff
@@ -210,13 +294,86 @@ StoragePool::~StoragePool()
 
 void StoragePool::enableGC()
 {
-    if (owned_storage)
+    if (run_mode == PageStorageRunMode::ONLY_V2 || run_mode == PageStorageRunMode::MIX_MODE)
+    {
         gc_handle = global_context.getBackgroundPool().addTask([this] { return this->gc(global_context.getSettingsRef()); });
+    }
+}
+
+void StoragePool::dataRegisterExternalPagesCallbacks(const ExternalPageCallbacks & callbacks)
```

Comment:
The callback may combine in `mixmode`. Or we will lose some data when GC happened Because V2 and V3 won't do GC at both times.